iterator.go

```diff
@@ -42,18 +52,11 @@ func (i *iterator) Next() (bool, error) {
 		return false, err
 	}
 
-	i.record = &record{
-		keySectionFactory: &sectionReaderFactory{
-			reader:   i.cdbReader.reader,
-			position: i.position + 8,
-			size:     keySize,
-		},
-		valueSectionFactory: &sectionReaderFactory{
-			reader:   i.cdbReader.reader,
-			position: i.position + 8 + keySize,
-			size:     valSize,
-		},
-	}
+	i.record.keySectionFactory.position = i.position + 8
+	i.record.keySectionFactory.size = keySize
```

Comment:
Unfortunately, it breaks the current iterator's behaviour:
`record` field is a pointer, and it means that each `Next` call modifies the same object again and again.
Please take a look at this test case, it works in the previous version:

```go
func (suite *CDBTestSuite) TestIteratorNext() {
	suite.fillTestCDB()
	iterator := suite.getCDBIterator()

	record1 := iterator.Record()
	suite.EqualRecords(record1, suite.testRecords[0])

	_, _ = iterator.Next()
	record2 := iterator.Record()
	suite.EqualRecords(record2, suite.testRecords[1])

	suite.EqualRecords(record1, suite.testRecords[0])
}
```

Reply:
What if I link `KeyBytes` and `ValueBytes`  to `iterator` structure (and rename them). Like this:

```go
 func (r *iterator) Key() ([]byte, error) {
...
```

```go
 func (r *iterator) Value() ([]byte, error) {
...
```

And `Record` method will return a copy of current record?
```go
// Record returns copy of current record
func (i *iterator) Record() Record {
	return &record{
		keySectionFactory: &sectionReaderFactory{
			reader:   i.record.keySectionFactory.reader,
			position: i.record.keySectionFactory.position,
			size:     i.record.keySectionFactory.size,
		},
		valueSectionFactory: &sectionReaderFactory{
			reader:   i.record.valueSectionFactory.reader,
			position: i.record.valueSectionFactory.position,
			size:     i.record.valueSectionFactory.size,
		},
	}
}
```

In this case number of allocations to get record with `Record()` method will not be changed.
And with `iterator.Key()` and `iterator.Value()` methods we will be able to get key or record with reduced number of allocations.

Reply:
This sounds good! We could mention, that `Record` method is lazy. Means, it will read data on require.

Reply:
You can use https://pkg.go.dev/golang.org/x/tools/cmd/benchcmp?tab=doc for comparing benchmarks

Reply:
> In this case number of allocations to get record with Record() method will not be changed.

Yes, it is, but you will reduce the number of allocation for `Next` method, because we will allocate memory for `Record` call. Good stuff:)

Reply:
I added your test now it passed. `Key()` `Value()` methods were added for `Iterator` interface